### ISSUE TITLE
Festivals don't occur, treated as normal days

### ISSUE THREAD

**Issue by mpql** [Pathoschild/SMAPI#484]
As title says. In multi-player, loaded up the first festival, the egg festival -- non-host without SMAPI entered egg festival, I (host) entered a non-festival village.

**Comment by Pathoschild** [comment]
This has also been [reported a few times in the forums](https://community.playstarbound.com/threads/smapi-stardew-modding-api.108375/page-118#post-3270024).

**Comment by mpql** [comment]
I saw lots in the 1.3 beta thread, but I didn't know all the circumstances surrounding their issues, and I didn't want to take ownership of them.

**Comment by Pathoschild** [comment]
Found the issue. The problem happens in the game's `Event.tryToLoadFestival` method:
```cs
public bool tryToLoadFestival(string festival)
{
    Game1.player.festivalScore = 0;
    try
    {
        this.festivalData = this.festivalContent.Load<Dictionary<string, string>>("Data\\Festivals\\" + festival);
        this.festivalData.Add("file", festival); 
    }
    catch (Exception ex)
    {
        return false;
    }
```

That method is called each time you enter a new location. Since SMAPI caches the asset, the `file` key is already set on subsequent loads and the `.Add` call crashes. The exception gets caught, and tryToLoadFestival always returns false.

I suggested changing that line to `this.festivalData["file"] = festival`, which will fix the issue with SMAPI and have no impact on the default code.

**Comment by Snipermonkey** [comment]
what file is it that I would have to edit to make this change to make this fix?

**Comment by Pathoschild** [comment]
The festival issue is fixed in the latest Stardew Valley 1.3.5 beta, and a more general fix will be done through #488.
